File: Py/c2_unidad1/P.O.O/Batalla.py
# Nesesario para crear los numeros aleatorios
import random
# Sirve para crear la barra de vida de los personajes 
from time import sleep
from tqdm import tqdm
# Genera los valores aleatorios para el daño de los personajes 
numram = random.randint(1,100)
numram2 = random.randint(1,100)
# clase del jugador
class Player:
  def __name__(self, name, arma, genero):
    self.name = name
    self.hp = 100
    self.attack = numram
    self.arma = arma
    self.genero = genero

# clase del enemigo
class Enemy:
  def __init__(self, name, arma, genero):
    self.name = name
    self.hp = 100
    self.attack = numram2
    self.arma = arma
    self.genero = genero

# función para simular la pelea
def fight(player, enemy):
  print(player.name + ' se enfrenta a ' + enemy.name)
  print('')
  # Mientras la vida de cualquiera de los 2 
  while player.hp > 0 and enemy.hp > 0:
    player_attack = random.randint(0, player.attack)
    enemy_attack = random.randint(0, enemy.attack)
    player.hp -= enemy_attack
    enemy.hp -= player_attack
    print('')
    print(player.name + ' ataca a ' + enemy.name + ' con: ',player.arma, 'y le quita ' + str(player_attack) + ' puntos de vida')
    print('')
    print('=== Vida del Kevin ===')
    # Estas son las lineas nesesarias para la creacion de la barra de vida progreciva 
    for i in tqdm(range(100)):
      sleep(0.00000001)
      if( i == enemy.hp):
        break
    print('')
    print(enemy.name + ' ataca a ' + player.name + ' con: ',enemy.arma,' y le quita ' + str(enemy_attack) + ' puntos de vida')
    print('')
    print('=== Vida del Sergio ===')
    for i in tqdm(range(100)):
        sleep(0.00000001)
        if( i == player.hp):
            break
  # Si cualquiera de los 2 es <= 0  entonces termina la similacion e imprime el mensaje 
  if player.hp <= 0:
    print('')
    print(player.name + ' ha sido derrotado')
  elif enemy.hp <= 0:
    print('')
    print(enemy.name + ' ha sido derrotado')

# Asignacion de los a tributos a los jugadores
player = Player('El Sergio', 'navaja', 'Hombre')
enemy = Enemy('El kevin', 'pistola', 'hombre')

# inicia la pelea
fight(player, enemy)

File: Jv/P.O.O/Batalla.java
import java.util.Random;
public class Batalla{
    //Clase del jugador 
    static public class Luchador{
        //Generador de los numeros aleatorios para el jugador 
        Random numAleatorio = new Random();
        int numram1 = numAleatorio.nextInt(100);
        private int vida;
        private int ataque;
        //Declarando los atributos del jugador
        public Luchador(int vida){
                this.vida = vida;
                this.ataque = numram1;
        }
    }
    //Clase del enemigo con herencia 
    static public class Karateca extends Luchador{
        //Generador de los numeros aleatorios para el enemigo
        Random numAleatorio = new Random();
        int numram2 = numAleatorio.nextInt(100);
        private static String nombre;
        private static int vida;
        private static int ataque;
        //Declarando los atributos del enemigo
        public Karateca(int vida, String nombre, String estilo){
            super(vida);
            this.nombre = nombre;
            this.vida = vida;
            this.ataque = numram2;
        }
    }  
    static public class Boxeador extends Luchador{
        //Generador de los numeros aleatorios para el enemigo
        Random numAleatorio = new Random();
        int numram2 = numAleatorio.nextInt(100);
        private static String nombre;
        private static int vida;
        private static int ataque;
        //Declarando los atributos del enemigo
        public Boxeador(int vida, String nombre, String estilo){
            super(vida);
            this.nombre = nombre;
            this.vida = vida;
            this.ataque = numram2;
        }   
    }
    //Inicia las simulacion de la batalla hasta que uno de los 2 llega a 0
    public static void main(String args[]){
        //Asiganamos los atributos a los persanajes 
        Luchador Sergio = new Karateca(100, "Sergio", "Karate");
        Luchador Kevin = new Boxeador(100, "Kevin", "Boxeo");
        System.out.println(Karateca.nombre + " se enfrenta a " + Boxeador.nombre);
        System.out.println("");
        //Mientras la vida de cualquiera de los 2 sea > 0 entonces segira repitiendo el bucle 
        while (Karateca.vida > 0 && Boxeador.vida > 0){
            int Player_Attack = Karateca.ataque;
            int Enemy_Attack = Boxeador.ataque;
            Karateca.vida -= Enemy_Attack;
            Boxeador.vida -= Player_Attack;
            System.out.println("");
            System.out.println(Karateca.nombre+" Ataca a Kevin y le quita " + Player_Attack + " puntos de vida");
            System.out.println("");
            System.out.println("=== Vida del Kevin ===");
            System.out.println(Boxeador.vida+"/100");
            System.out.println("");
            System.out.println(Boxeador.nombre+" Ataca a Sergio y le quita " + Enemy_Attack + " puntos de vida");
            System.out.println("");
            System.out.println("=== Vida del Sergio ===");
            System.out.println(Karateca.vida+"/100");
            //Si cualquira de los 2 llega a 0 entonces la simulacion termina e imprime el mensaje 
            if (Karateca.vida <= 0){
                System.out.println("");
                System.out.println("Sergio a sido derrotaro");
                System.out.println("");
            }
            else if (Boxeador.vida <= 0){
                System.out.println("");
                System.out.println("Kevin a sido derrotaro");
                System.out.println("");
            }
        }
    }
}